File: lyedu-api/src/main/java/com/lyedu/entity/ExamRecord.java
package com.lyedu.entity;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 考试记录实体（用户交卷）
 *
 * @author LyEdu Team
 */
@Data
public class ExamRecord {

    private Long id;
    private Long examId;
    private Long userId;
    private Long paperId;
    /** 得分 */
    private Integer score;
    /** 是否及格：0-否，1-是 */
    private Integer passed;
    /** 用户答案 JSON */
    private String answers;
    private LocalDateTime submitTime;
    private LocalDateTime createTime;
}

File: lyedu-admin/src/api/examRecord.ts
import request from '@/utils/request'

export interface ExamRecord {
  id: number
  examId: number
  examTitle?: string
  userId: number
  realName?: string
  username?: string
  paperId: number
  score: number
  passed: number
  answers?: string
  submitTime?: string
  createTime?: string
}

export interface PageResult<T> {
  records: T[]
  total: number
}

export const getExamRecordPage = (params: {
  page: number
  size: number
  keyword?: string
  examId?: number
  userId?: number
}) => {
  return request.get<PageResult<ExamRecord>>('/exam-record/page', { params })
}
